plugin/src/test/java/org/opensearch/ml/rest/RestMLRemoteInferenceIT.java

```diff
@@ -237,6 +237,70 @@ public void testPredictRemoteModel() throws IOException, InterruptedException {
         assertFalse(((String) responseMap.get("text")).isEmpty());
     }
 
+    public void testPredictRemoteModelWithInterface(String testCase, Consumer<Map> verifyResponse, Consumer<Exception> verifyException)
+        throws IOException,
+        InterruptedException {
+        // Skip test if key is null
+        if (OPENAI_KEY == null) {
+            return;
+        }
+        Response response = createConnector(completionModelConnectorEntity);
+        Map responseMap = parseResponseToMap(response);
+        String connectorId = (String) responseMap.get("connector_id");
+        response = registerRemoteModelWithInterface("openAI-GPT-3.5 completions", connectorId, testCase);
+        responseMap = parseResponseToMap(response);
+        String taskId = (String) responseMap.get("task_id");
+        waitForTask(taskId, MLTaskState.COMPLETED);
+        response = getTask(taskId);
+        responseMap = parseResponseToMap(response);
+        String modelId = (String) responseMap.get("model_id");
+        response = deployRemoteModel(modelId);
+        responseMap = parseResponseToMap(response);
+        taskId = (String) responseMap.get("task_id");
+        waitForTask(taskId, MLTaskState.COMPLETED);
+        String predictInput = "{\n" + "  \"parameters\": {\n" + "      \"prompt\": \"Say this is a test\"\n" + "  }\n" + "}";
+        try {
+            response = predictRemoteModel(modelId, predictInput);
+            responseMap = parseResponseToMap(response);
+            verifyResponse.accept(responseMap);
+        } catch (Exception e) {
+            verifyException.accept(e);
```

Comment:
wondering if can check on the model interface in the response?

Reply:
What do you mean by "check on the model interface in the response"

Reply:
you can try verify the response for connector_id and model_id

https://github.com/opensearch-project/ml-commons/blob/8cb6a9b6de485f57bf0b3a8e8ea9905aa292ad0f/plugin/src/test/java/org/opensearch/ml/rest/MLCommonsRestTestCase.java#L633